Sources/NIO/Socket.swift

```diff
@@ -174,14 +174,27 @@ typealias IOVector = iovec
 
         return try withUnsafeHandle { handle in
             return try withUnsafeMutablePointer(to: &vec) { vecPtr in
+#if os(Windows)
+                var messageHeader =
+                    WSAMSG(name: notConstCorrectDestinationPtr
+                                    .assumingMemoryBound(to: SOCKADDR.self),
+                           namelen: destinationSize,
+                           lpBuffers: vecPtr,
+                           dwBufferCount: 1,
+                           Control: WSABUF(len: ULONG(controlBytes.count),
+                                           buf: controlBytes.baseAddress?
+                                                    .assumingMemoryBound(to: CHAR.self)),
```

Comment:
This `assumingMemoryBound` is a bit rough. It's technically a violation of Swift's pointer aliasing model, though it's ok in C.

Normally the way we'd address this is to thunk through CNIOWindows by having it declare the API with the types we want and just doing the cast that is unsafe in Swift. However, in this case I think it's acceptable for us to bind the memory: the rest of the program assumes it's untyped, so we won't break the aliasing model. Can we replace the `assumingMemoryBound` here and below with `.bindMemory(to:)`?